Updates to support Verizon Ads SDK v1.1.1

nimrich opened the pull request.

chauduyphanvu1 commented on Verizon/src/main/java/com/mopub/mobileads/VerizonInterstitial.java:
```diff
@@ -69,39 +71,37 @@ protected void loadInterstitial(final Context context,
-        final String placementId = serverExtras.get(getPlacementIdKey());
+        // Ensure that siteId is the key and cache serverExtras so siteId can be used to initialize VAS early at next launch
+        if (!TextUtils.isEmpty(siteId)) {
+            serverExtras.put(VerizonAdapterConfiguration.VAS_SITE_ID_KEY, siteId);
```
Isn't the site ID String returned from the MoPub ad server? If so, it's always going to be in `serverExtras` that you can cache (by calling `verizonAdapterConfiguration.setCachedInitializationParameters(context, serverExtras)`) right away. There's no need to put another entry for site ID in the Map.

chauduyphanvu1 commented on Verizon/src/main/java/com/mopub/mobileads/VerizonInterstitial.java:
```diff
@@ -69,39 +71,37 @@ protected void loadInterstitial(final Context context,
 
-            if (!success) {
-                MoPubLog.log(CUSTOM, ADAPTER_NAME, "Failed to initialize the Verizon SDK");
+            if ((application != null) && (!StandardEdition.initialize(application, siteId))) {
```
Do you mean to check if `application == null` here? Also, remove extra parentheses.

chauduyphanvu1 commented on Verizon/src/main/java/com/mopub/mobileads/VerizonInterstitial.java:
```diff
@@ -69,39 +71,37 @@ protected void loadInterstitial(final Context context,
-
-                return;
+        if (!VASAds.isInitialized()) {
+            Application application = null;
```
Add new line below.

chauduyphanvu1 commented on Verizon/src/main/java/com/mopub/mobileads/VerizonBanner.java:
```diff
@@ -75,39 +77,34 @@ protected void loadBanner(final Context context,
-            }
+        // Ensure that siteId is the key and cache serverExtras so siteId can be used to initalizate VAS early at next launch
+        if (!TextUtils.isEmpty(siteId)) {
+                serverExtras.put(VerizonAdapterConfiguration.VAS_SITE_ID_KEY, siteId);
```
Same comment regarding putting another site ID entry to this Map.

chauduyphanvu1 commented on Verizon/src/main/java/com/mopub/mobileads/VerizonBanner.java:
```diff
@@ -75,39 +77,34 @@ protected void loadBanner(final Context context,
-            if (!(context instanceof Activity)) {
-                MoPubLog.log(CUSTOM, ADAPTER_NAME, "Ad request to Verizon failed because " +
-                        "context is not an Activity");
+            if ((application != null) && (!StandardEdition.initialize(application, siteId))) {
```
if `application == null`?

chauduyphanvu1 commented on Verizon/src/main/java/com/mopub/mobileads/VerizonAdapterConfiguration.java:
```diff
@@ -60,17 +65,30 @@ public void initializeNetwork(@NonNull final Context context, @Nullable final Ma
+                if ((!TextUtils.isEmpty(finalSiteId)) && (context instanceof Application) &&
+                    (StandardEdition.initialize((Application) context, finalSiteId))) {
+
+                    listener.onNetworkInitializationFinished(VerizonAdapterConfiguration.class,
```
Please also notify `listener.onNetworkInitializationFinished()` with the `MoPubErrorCode.ADAPTER_CONFIGURATION_ERROR` error code when the if conditional on line 77 isn't `true`.

nimrich commented on Verizon/src/main/java/com/mopub/mobileads/VerizonAdapterConfiguration.java:
```diff
@@ -60,17 +65,30 @@ public void initializeNetwork(@NonNull final Context context, @Nullable final Ma
+            @Override
+            public void run() {
+                if ((!TextUtils.isEmpty(finalSiteId)) && (context instanceof Application) &&
+                    (StandardEdition.initialize((Application) context, finalSiteId))) {
```
It does initialize the SDK.

nimrich commented on Verizon/src/main/java/com/mopub/mobileads/VerizonBanner.java:
```diff
@@ -75,39 +77,34 @@ protected void loadBanner(final Context context,
+        // The current activity must be set as resumed so VAS can track ad visibility
+        ActivityStateManager activityStateManager = VASAds.getActivityStateManager();
+        if ((activityStateManager != null) && (context instanceof Activity)) {
+                activityStateManager.setState((Activity) context, ActivityStateManager.ActivityState.RESUMED);
```
The access level was modified in v1.1.1 of the Verizon SDK.

nimrich commented on Verizon/src/main/java/com/mopub/mobileads/VerizonInterstitial.java:
```diff
@@ -69,39 +71,37 @@ protected void loadInterstitial(final Context context,
+        // The current activity must be set as resumed so VAS can track ad visibility
+        ActivityStateManager activityStateManager = VASAds.getActivityStateManager();
+        if ((activityStateManager != null) && (context instanceof Activity)) {
+            activityStateManager.setState((Activity) context, ActivityStateManager.ActivityState.RESUMED);
```
The access level was modified in v1.1.1 of the Verizon SDK.

nimrich commented:
@ chauduyphanvu1 I believe I have addressed all of your comments.  Thanks so much for the quick turnaround on the review!

Just let me know if there's anything else I missed!

nimrich commented on Verizon/src/main/java/com/mopub/mobileads/VerizonAdapterConfiguration.java:
```diff
@@ -60,17 +65,40 @@ public void initializeNetwork(@NonNull final Context context, @Nullable final Ma
+                if (mopubLogLevel == MoPubLog.LogLevel.DEBUG) {
+                    VASAds.setLogLevel(Logger.DEBUG);
+                } else if (mopubLogLevel == MoPubLog.LogLevel.INFO) {
+                    VASAds.setLogLevel(Logger.INFO);
```
It does not need to be initialized first so I'll go ahead and move it.

nimrich commented:
@chauduyphanvu1 went ahead and made that change.
